core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/principal/Principal.scala

```diff
@@ -76,7 +76,7 @@ class Principal(val metadata: OpExecConfig)
   implicit val ec: ExecutionContext = context.dispatcher
   implicit val timeout: Timeout = 5.seconds
 
-  lazy val promiseHandlerInitializer = wire[PromiseHandlerInitializer]
+  lazy val rpcHandlerInitializer = wire[RPCHandlerInitializer]
```

Comment:
again, why do we have this if the workflowactor already has it

Reply:
WorkflowActor's rpcHandlerInitializer is abstract.